substrate-node/pallets/pallet-tfgrid/src/grid_migration.rs

```diff
@@ -243,15 +243,32 @@ pub fn migrate_farms<T: Config>() -> frame_support::weights::Weight {
             }
         }
 
-        let replaced_farm_name = farm::replace_farm_name_spaces_with_underscores(&farm.name);
-        let name = match <T as Config>::FarmName::try_from(replaced_farm_name) {
+        let mut farm_name = farm.name;
+        let truncated = farm_name.len() > <T as Config>::MaxFarmNameLength::get() as usize;
+        if truncated {
+            info!("farm name length exceeded, truncating farm name");
+            farm_name.truncate(<T as Config>::MaxFarmNameLength::get() as usize);
+        }
+
+        let replaced_farm_name = farm::replace_farm_name_invalid_characters(&farm_name);
+        if replaced_farm_name != farm_name || truncated {
+            info!("farm name changed, removing from farm id by name map");
+            FarmIdByName::<T>::remove(&farm_name);
+        }
+
+        let name = match <T as Config>::FarmName::try_from(replaced_farm_name.clone()) {
             Ok(n) => n,
             Err(_) => {
                 info!("invalid farm name, skipping updating farm {:?} ...", k);
                 return None;
             }
         };
 
+        if replaced_farm_name != farm_name || truncated {
+            info!("farm name changed, inserting into farm id by name map");
+            FarmIdByName::<T>::insert(replaced_farm_name, farm.id);
+        }
```

Comment:
There is 2 identical if checks here, one removes the entry and the other adds the changed entry. This should be merged into 1 block